Cataclysm/Druid.lua

```diff
@@ -662,39 +696,34 @@ spec:RegisterStateExpr("calc_bite_dpe", function()
     local avg_base_damage = 377.877920772
     local scaling_per_combo_point = 0.125
     local dmg_per_combo_point = 576.189844175
+    local cp = combo_points.current
 
     local base_cost = (buff.clearcasting.up and 0) or (25 * ((buff.berserk.up and 0.5) or 1))
     local excess_energy = min(25, energy.current - base_cost) or 0
 
-    local bite_damage =         avg_base_damage
-    bite_damage = bite_damage + dmg_per_combo_point * combo_points.current 
-    bite_damage = bite_damage + state.stat.attack_power * scaling_per_combo_point * combo_points.current
+    local bonus_crit = rend_and_tear_mod_shred
+    local damage_multiplier = (1 + talent.feral_aggression.rank*0.05) * (1 + excess_energy/25)
 
-    -- Hekili:Debug("bite_damage pre dmgc (%.2f), attPower (%d)", bite_damage, state.stat.attack_power)
-    -- bite_damage = calculate_damage(1, bite_damage, false, false, false, false) TODO fix
-    Hekili:Debug("bite_damage (%.2f), excess_energy (%d)", bite_damage, excess_energy)
+    local bite_damage = avg_base_damage + cp*(dmg_per_combo_point + state.stat.attack_power*scaling_per_combo_point)
 
-    bite_damage = bite_damage * (1 + excess_energy/25)
-    bite_damage = bite_damage * (1 + talent.feral_aggression.rank * 0.05)
-    
-    Hekili:Debug("bite_damage final (%.2f), excess_energy (%d)", bite_damage, excess_energy)
+    bite_damage = calculate_damage(bite_damage, 0, false, false, true, damage_multiplier, bonus_crit) 
+    Hekili:Debug("bite_damage (%.2f), excess_energy (%d)", bite_damage, excess_energy)
 
-    local bite_dpe = bite_damage / (base_cost + excess_energy)
+    local bite_dpe = bite_damage / (base_cost + excess_energy) -- TODO: check if this should include excess energy
 
     return bite_dpe
 end)
 
-spec:RegisterStateExpr("calc_rip_tick_damage", function()
+spec:RegisterStateExpr("calc_rip_tick_damage", function() -- TODO move this to an action?
```

Comment:
what do you mean with "to an action"?

Reply:
Most properties associated with a skill have an _aura_ and _action_, these function as an object to encapsulate the behavior of the statistics or use the skill, it is not normal to have separate functions to calculate damage, maybe we can be move this to an `action.rip.tick-damage`

Reply:
thats what you ment, yes I know this concept^^
I just did it like that for making it easier to compare to the wowsim implementation